pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -1332,6 +1423,212 @@ func TestUserSignupDeactivatedAfterMURCreated(t *testing.T) {
 		require.Equal(t, userSignup.Name, notification.Spec.UserID)
 		assert.Equal(t, "userdeactivated", notification.Spec.Template)
 	})
+
+}
+
+func TestUserSignupFailedToCreateDeactivationNotification(t *testing.T) {
+	// given
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: newObjectMeta("", "john.doe@redhat.com"),
+		Spec: v1alpha1.UserSignupSpec{
+			Username:    "john.doe@redhat.com",
+			Deactivated: true,
+		},
+		Status: v1alpha1.UserSignupStatus{
+			Conditions: []v1alpha1.Condition{
+				{
+					Type:   v1alpha1.UserSignupComplete,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1alpha1.UserSignupApproved,
+					Status: v1.ConditionTrue,
+					Reason: "ApprovedAutomatically",
+				},
+			},
+			CompliantUsername: "john-doe",
+		},
+	}
+	userSignup.Labels["toolchain.dev.openshift.com/approved"] = "true"
+	key := test.NamespacedName(operatorNamespace, userSignup.Name)
+
+	t.Run("when the deactivation notification cannot be created", func(t *testing.T) {
+		// given
+		InitializeCounter(t, 2)
+		defer counter.Reset()
+		r, req, cl := prepareReconcile(t, userSignup.Name, userSignup, configMap(configuration.UserApprovalPolicyAutomatic), basicNSTemplateTier)
+
+		cl.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+			switch obj.(type) {
+			case *v1alpha1.Notification:
+				return errors.New("unable to create deactivation notification")
+			default:
+				return cl.Create(ctx, obj)
+			}
+		}
+
+		// when
+		_, err := r.Reconcile(req)
+
+		// then
+		require.Error(t, err)
+		require.Equal(t, "Failed to create user deactivation notification: unable to create deactivation notification", err.Error())
+
+		// Lookup the UserSignup
+		err = r.client.Get(context.TODO(), key, userSignup)
+		require.NoError(t, err)
+
+		// Confirm the status has been set to Deactivated
+		test.AssertConditionsMatch(t, userSignup.Status.Conditions,
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupApproved,
+				Status: v1.ConditionTrue,
+				Reason: "ApprovedAutomatically",
+			},
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupComplete,
+				Status: v1.ConditionTrue,
+			},
+			v1alpha1.Condition{
+				Type:    v1alpha1.UserSignupUserDeactivatedNotificationCreated,
+				Status:  v1.ConditionFalse,
+				Reason:  "NotificationCRCreationFailed",
+				Message: "unable to create deactivation notification",
+			})
+		AssertThatCounterHas(t, 2)
+
+		// A deactivated notification should have been created
```

Comment:
It should or it shouldn't? You assert that it's not created.

Reply:
Oops, should be "shouldn't". Fixed in https://github.com/codeready-toolchain/host-operator/pull/289/commits/448e505672962faf3eda38deaa43deb1de12d8c7